### PR TITLE
Add initial recipe for py2app

### PR DESCRIPTION
**Pull request by bkpoon**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
Can you try adding `libgcc` for run ?

**Comment by bkpoon** [comment]
@jan-janssen Adding `libgcc` didn't work either. This package has prebuilt binaries for building an app bundle for macOS. The `apptemplate` directory has prebuilt binaries for multiple architectures and seems to be used by the `py2app/build_app.py` and `py2app/create_appbundle.py` scripts, and the `bundletemplate` directory has prebuilt binaries that are used by `py2app/create_pluginbundle.py`. These 2 approaches seem to be the way to bundle standard Python packages built with `setuptools`, which is why I was trying to get the recipe to ignore these prebuilt binaries when looking for shared libraries. The 3rd way to use this package is `py2applet`, which is what I use to bundle a wxPython app in a conda environment.

A second issue with this package is that this is really only useful for macOS, which is why I have the `skip: true  # [not osx]` line in the `build` section. However, there's no reason to build this package for different versions of Python, but the linter did not like the `noarch: python` line with the selector. So is there a `conda-forge` policy about `noarch: python` packages that are only useful for one platform? I can change to recipe to reflect the policy.

Thanks!

**Comment by bkpoon** [comment]
@conda-forge/help-python-c
Checking in to see if there are any suggestions for getting the prebuilt binaries in the source package to be ignored? Thanks!

**Comment by bkpoon** [comment]
@conda-forge-admin, please rerender

**Comment by bkpoon** [comment]
@conda-forge/staged-recipes This took a little longer than I expected, but I think this is ready for review.

Two issues,

1) I added `libgcc_s.1.dylib` to `missing_dso_whitelist` to prevent the build from failing. The image for building does not seem to have `/usr/lib/libgcc_s.1.dylib`, so `conda-build` would fail when trying to update the links in the prebuilt binaries. I also added a script that adds RPATH to that library.
2) This package is specific to macOS, so I added `skip: true  [not osx]`, but it does not really need separate packages for 2.7, 3.6, etc. Would `noarch: python` be better?

Thanks!

File: recipes/py2app/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  entry_points:
+    - py2applet = py2app.script_py2applet:main
+  missing_dso_whitelist:
+    - '*/libgcc_s.1.dylib'
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
```
Comment (scopatz) [inline recipes/py2app/meta.yaml]: please remove setuptools, since pip is a dep

File: recipes/py2app/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  run:
+    - altgraph >=0.16
+    - macholib >=1.11
+    - modulegraph >=0.17
+    - python
+    - setuptools
```
Comment (scopatz) [inline recipes/py2app/meta.yaml]: is setuptools really needed at run time by the user??

**Comment by scopatz** [comment]
so noarch would be good, except that

**Comment by scopatz** [comment]
So noarch would be good except that the way selectors work, you wouldn't be able to restrict it to mac only

File: recipes/py2app/meta.yaml (hunk 1 above)
Comment (bkpoon) [reply recipes/py2app/meta.yaml]: Fixed!

File: recipes/py2app/meta.yaml (hunk 2 above)
Comment (bkpoon) [reply recipes/py2app/meta.yaml]: One way to use this package is to build `pip` packages that are then put into a .app bundle for macOS, so `setuptools` is needed for that functionality (https://py2app.readthedocs.io/en/latest/dependencies.html). I'm using this package another way that can use `conda` environments in .app bundles.

**Comment by bkpoon** [comment]
I have no preference, so keeping the selector is fine with me. I just wanted to double-check if there was a preferred way for platform-specific packages that don't need separate Python builds.

**Pull request merged**